src/task/task.cpp

```diff
@@ -1,51 +1,142 @@
 #include "task/task.h"
-#include "task/task_manager.h"
+
+#include <utility>
 
 #include "optimizer/cost_model/trivial_cost_model.h"
+#include "parser/expression/constant_value_expression.h"
+#include "task/task_manager.h"
 #include "util/query_exec_util.h"
 
 namespace noisepage::task {
 
+Task::Task(const catalog::db_oid_t db_oid, std::string &&query_text, const transaction::TransactionPolicy policy,
+           const common::ManagedPointer<common::FutureDummy> sync)
+    : db_oid_(db_oid), query_text_(std::move(query_text)), policy_(policy), sync_(sync) {}
+
+std::unique_ptr<TaskDDL> TaskDDL::Builder::Build() {
+  NOISEPAGE_ASSERT(db_oid_.has_value(), "Database OID must be specified.");
+  NOISEPAGE_ASSERT(*db_oid_ != catalog::INVALID_DATABASE_OID, "You're doing something hacky. Get a real database OID.");
+  NOISEPAGE_ASSERT(query_text_.has_value(), "Query text must be specified.");
+  NOISEPAGE_ASSERT(policy_.has_value(), "Transaction policy must be specified.");
+  return std::unique_ptr<TaskDDL>(new TaskDDL(*db_oid_, std::move(*query_text_), *policy_, sync_));
+}
+
+std::unique_ptr<TaskDML> TaskDML::Builder::Build() {
+  // Sanity checks.
+  NOISEPAGE_ASSERT(db_oid_.has_value(), "Database OID must be specified.");
+  NOISEPAGE_ASSERT(*db_oid_ != catalog::INVALID_DATABASE_OID, "You're doing something hacky. Get a real database OID.");
+  NOISEPAGE_ASSERT(query_text_.has_value(), "Query text must be specified.");
+  NOISEPAGE_ASSERT(policy_.has_value(), "Transaction policy must be specified.");
+  NOISEPAGE_ASSERT(!override_qid_.has_value() || should_skip_query_cache_,
+                   "Cannot override query ID without skipping the query cache.");
+  NOISEPAGE_ASSERT(params_.empty() || (params_.at(0).size() == param_types_.size()),
+                   "Parameter types do not match parameters in length.");
+  // Set defaults.
+  cost_model_ = cost_model_ != nullptr ? std::move(cost_model_) : std::make_unique<optimizer::TrivialCostModel>();
+
+  return std::unique_ptr<TaskDML>(new TaskDML(*db_oid_, std::move(*query_text_), *policy_, sync_,
+                                              std::move(cost_model_), std::move(params_), std::move(param_types_),
+                                              settings_, override_qid_, metrics_manager_, should_force_abort_,
+                                              should_skip_query_cache_, std::move(tuple_fn_)));
+}
+
+TaskDML::Builder &TaskDML::Builder::SetCostModel(std::unique_ptr<optimizer::AbstractCostModel> cost_model) {
+  cost_model_ = std::move(cost_model);
+  return *this;
+}
```

Comment:
Mega-nit: empty lines between these function definitions.

Reply:
I'm surprised format didn't do it, actually.. done!